carma_wm/include/carma_wm/Geometry.h

```diff
@@ -141,6 +141,31 @@ double getAngleBetweenVectors(const Eigen::Vector2d& vec1, const Eigen::Vector2d
  * frame and moving clockwise
  */
 lanelet::BasicPolygon2d objectToMapPolygon(const geometry_msgs::Pose& pose, const geometry_msgs::Vector3& size);
+
+/*! \brief Helper function to identify whether to select the preceeding or succeeding segment of a linstring point
+   * that is nearest an external point when trying to find the TrackPos of the external point. Of the 3 points
+   * comprising the 2 segment linestirng the external point must be closest to the midpoint for this function to be
+   * valid
+   *
+   * ASSUMPTION: Of the 3 points comprising the 2 segment linestirng the external point is closest to the midpoint
+   *
+   * Meant to be called in the matchSegment method. Logic is as follows
+   * If the external point is within the downtrack bounds of one segment but not the other then use the one it is within
+   * If the external point is within the downtrack bounds of both segments then use the one with the smallest crosstrack
+   * distance If the external point is within the downtrack bounds of both segments and has exactly equal crosstrack
+   * bounds with each segment then use the preceeding segment If the external point is outside the downtrack bounds of
+   * both segments then assign to the first segment as it will always have positive downtrack in this case while the
+   * second segment will always have negative downtrack
+   *
+   * \param first_seg_trackPos The TrackPos of the external point relative to the preceeding segment
+   * \param second_seg_trackPos The TrackPos of the succeeding segment
+   * \param first_seg_length The length of the preceeding segment
+   * \param second_seg_length The length of the succeeding segment
+   *
+   * \return True if the preceeding segment is the better choice. False if the succeeding segment is the better choice.
+   */
+  bool selectFirstSegment(const TrackPos& first_seg_trackPos, const TrackPos& second_seg_trackPos,
```

Comment:
I'm still not clear on why this function was exposed. Its a very specific use case for it and you don't appear to be using it anywhere.

Reply:
This is mistake sorry. This is somehow keep being reversed